packages/repository/src/define-model-class.ts

```diff
@@ -60,6 +61,9 @@ export function defineModelClass<
     Props
   >;
   assert.equal(modelClass.name, modelName);
+
+  // Apply `@model(definition)` to the generated class
+  model(definition)(modelClass);
   modelClass.definition = definition;
```

Comment:
Do we still need to set `modelClass.definition`? I believe `@model` decorator handles that part automatically.

Reply:
I just commented the line and ran the tests, three of them fail. So looks like there's more to it.

Reply:
Can you check why they are failing? The `@model` calls `buildModelDefinition()`, which should set `definition` to the class.

Reply:
Using `deepEqual` instead of `equal` fixed it.

Reply:
`equal` is for the same object.